src/layout/Table.js

```diff
@@ -0,0 +1,108 @@
+                {F.mapIndexed(
+                  ({ field, display = x => x, details = {} }, i) => (
+                    <td
+                      key={`${field}${i}`}
+                      {...!_.isEmpty(details) && {
+                        style: {
+                          cursor: !_.isEmpty(details) ? 'pointer' : 'auto',
+                        },
+                        onClick() {
+                          let indexedField = `${field}${i}`
+                          let expandedRow = state.expandedRows.find(
+                            row => row.key === x[recordKey]
+                          )
+                          if (!expandedRow) {
+                            state.expandedRows.push({
+                              key: x[recordKey],
+                              record: x,
+                              field,
+                              indexedField,
+                              details,
+                            })
+                          } else if (
+                            expandedRow.indexedField !== indexedField
+                          ) {
+                            expandedRow.field = field
+                            expandedRow.indexedField = indexedField
+                            expandedRow.details = details
+                          } else {
+                            state.expandedRows.remove(expandedRow)
+                          }
+                        },
+                      }}
+                    >
+                      {_.getOr(
+                        display,
+                        `${
+                          state.expandedRows.find(
+                            row =>
+                              row.key === x[recordKey] &&
+                              row.indexedField === `${field}${i}`
+                          )
+                            ? 'collapse'
+                            : 'expand'
+                        }.display`,
+                        details
+                      )(_.get(field, x), x)}
+                    </td>
+                  ),
+                  columns
+                )}
+              </tr>
+              {/* See if there is a details component to render for the column value when row expanded */}
+              {_.flow(
+                key => state.expandedRows.find(row => row.key === key),
+                expandedRow =>
+                  _.get('details.component', expandedRow) && (
+                    <tr>
+                      <td colSpan={columns.length}>
+                        {expandedRow.details.component(
+                          _.get(expandedRow.field, expandedRow.record),
+                          expandedRow.record
+                        )}
+                      </td>
+                    </tr>
+                  )
+              )(x[recordKey])}
+            </React.Fragment>
+          ),
+          data
+        )}
+      </tbody>
+    </table>
+  ))
+)
+
+Table.displayName = 'Table'
```

Comment:
I also wonder if the Results Table should use this Table component!

Reply:
This component doesn't have knowledge of contexture trees but I guess the ResultsTable could use it for rendering, not sure what the scope/effort of such refactoring would be.

Reply:
Yeah, me neither. I guess what I _really_ wonder is if we have cases of result nodes being rendered with expandable tables? Are you aware of similar use cases?

Reply:
So far no, but that sounds likely at some point.